Feature/deep first search

Zaldef opened the pull request:
POR FAVOR ANALISEM BEM O CODIGO ANTES DO MERGE, qualquer duvida me perguntem antes de aprovar, eliminei whitespaces e algumas pequenas alterações recomendadas pelo Pylint, uma extensao que ajuda seguir um padrao de codificaçào em python, se quiserem saber mais so entrar no site deles, ou ler a PEP8

Zaldef committed:
Develop (#19)

Zaldef committed:
Implementada Interface Grafica

Vitor-Yuzo committed:
Feature/GUI (#20)

Vitor-Yuzo committed:
codigo_principal

Falta incrementar os prints de resolução da IA, passo a passo

Vitor-Yuzo committed:
Merge remote-tracking branch 'origin/develop' into feature/A-estrela

Vitor-Yuzo committed:
codigo_principal_erro_main

Corrigir o erro na hora de mandar o estado como parametro. Recomendo, ou criar instanciar um estado na função, ou na main para que todas as outras funções possam usar

Zaldef committed:
Merge branch 'feature/Deep-first-search' into feature/A-estrela

Zaldef committed:
a* -> deep-first (#22)

Zaldef committed:
implementado deep-first

Zaldef committed:
Removido Warnings e erros

Pedro22D0W commented on main.py:
```diff
@@ -127,31 +126,139 @@ def validar_movimento(matriz, movimento):
     else:
         return False  # Movimento inválido
-    
-def jogada_usuario(estados, interface):
```
porque removeu?

Pedro22D0W commented on main.py:
```diff
@@ -127,31 +126,139 @@ def validar_movimento(matriz, movimento):
+                novo_estado = Estado(estado_atual.matriz, movimento)
+                matriz_str = str(novo_estado.matriz)  # Converter para string para verificar se foi visitado
+
+                if matriz_str not in visitados:
```
busca em profundidade precisa armazenar os estados visitados? isso não deveria ser A*?

Pedro22D0W reviewed (commented):
cu .

Zaldef committed:
Profundidade limitada

Modificada a def de busca_profundidade para explorar nossa "arvore" até a profundidade 50 dela

Zaldef committed:
recovery def jogada_usuario

em algum momento ela foi excluida, minha branch ja nasceu com ela excluida, deve ter sido originario da busca a*

Zaldef commented:
Novo PR em realização
